lib/filesystem/src/filesystem.cpp

```diff
@@ -723,6 +723,43 @@ namespace irods::experimental::filesystem::NAMESPACE_IMPL
         return 0;
     }
 
+    auto remove_all(rxComm& _comm, const path& _p, extended_remove_options _opts) -> std::uintmax_t
+    {
+        std::string data_obj_sql = "select count(DATA_ID) where COLL_NAME like '";
+        data_obj_sql += _p;
+        data_obj_sql += "%'";
+
+        std::string colls_sql = "select count(COLL_ID) where COLL_NAME like '";
+        colls_sql += _p;
+        colls_sql += "%'";
```

Comment:
fmt could make this more readable.  but, this is also fairly easy to understand.